core/src/main/java/com/opensymphony/xwork2/interceptor/ParametersInterceptor.java

```diff
@@ -354,7 +407,86 @@ protected boolean isExcluded(String paramName) {
         }
         return false;
     }
+    
 
+    public void setAcceptedValuePatterns(String commaDelimitedPatterns) {
```

Comment:
BTW another methods annotated with `@Inject` as well as some other changes are required enabling user to configure these patterns via XML configuration.

Reply:
No matter what I did I couldn't get Inject to work on these, and I had assumed that's because I was using it inappropriately. The other setters that are documented by Struts, setParamNameMaxLength, setAcceptParamNames, setExcludeParams don't utilize the @Inject annotation and are injected via the following params and thus I followed suit

acceptParamNames, excludeParams, paramNameMaxLength

Reply:
Ah I see , thanks for clarification!

Reply:
Hmm I think it makes sense to have some default patterns set here however. Currently it's empty and user has to set it in all interceptors stacks as far as I understood right? If so then like what is done for parameters names, better to have some defaults patterns set here e.g. your already provided patterns in tests for %{} and ${}

Reply:
My concern would be the regression impact. I would sure hope that Struts users aren't ever expecting users to pass in params with values with %{} and ${} but that level of change seems best to include with a major (which ironically I just missed).

I do think we could prevent an entire attack vector by excluding %{} and ${} in param values but again I'd be worried about the regression impact.

If the Struts team wants to do that I wouldn't object but I wasn't going to go that far on my own.

Reply:
And yes it would need to be defined such as:

    <package name="default" extends="struts-default">
        <interceptors>
            <interceptor-stack name="myCustomStack">
                <interceptor-ref name="defaultStack">
                	       <param name="params.excludeValuePatterns">.*\$\{.*?\}.*,.*%\{.*?\}.*</param>
                </interceptor-ref>
            </interceptor-stack>
        </interceptors>
  </package>